app-sre/qontract-reconcile
# Clean up Terraform working dirs on print to file

Temporary directories are not cleaned up when neither tf or working_dirs
kwargs are passed into cleanup_and_exit. This appears to affect print to
file functionality in development, but worth fixing. It's simple enough
to add a breakpoint if team members want to preserve the temporary dirs.

A prefix was added to the temporary directories as well so that it's easier
to identify which client is producing them. This will become increasingly
helpful when there is more than just the Terrascript AWS client.

Discovered as part of: https://issues.redhat.com/browse/APPSRE-5922

## Testing

* Run `terraform-resources` with `--print-to-file` and confirm that a temporary directory was created (`terrascript-aws-_doj_004`) and that it was deleted afterwards
* Run `terraform-resources` in `--dry-run` mode to verify that the change in temporary directory name didn't impact anything
